cmd/kubeadm/app/cmd/phases/init/waitcontrolplane.go

```diff
@@ -109,6 +109,16 @@ func runWaitControlPlanePhase(c workflow.RunData) error {
 		return errors.New("couldn't initialize a Kubernetes cluster")
 	}
 
+	waiter.SetTimeout(kubeadmconstants.DefaultKubeletTimeout)
```

Comment:
Do we support adjusting the waiting time for kubelet?
I am worried whether on a low-configuration machine, this wait will always time out.

Reply:
> Do we support adjusting the waiting time for kubelet?

no, if we plan to support it, it will be part of v1beta4.
maybe i can try taking ownership of that task:
https://github.com/kubernetes/kubeadm/issues/2463

> I am worried whether on a low-configuration machine, this wait will always time out.

the timeout before this change was also hardcoded by less - 40sec, now it's 1 minute.

Reply:
If I understand it correctly, the wait time should be even longer.
We used the backoff mode, first waiting about 40s before do kubelet check, then check 5 times in backoff mode(5s +10s +20s + 40s + 80s = 155s), so the total timeout should be 195s, about 3 minutes.

Reply:
Perhaps a safe timeout duration is 4 minutes until the custom configuration is supported. WDYT?

Reply:
> If I understand it correctly, the wait time should be even longer.
We used the backoff mode, first waiting about 40s before do kubelet check, then check 5 times in backoff mode(5s +10s +20s + 40s + 80s = 155s), so the total timeout should be 195s, about 3 minutes.

you are right, i forgot it was a backoff with the TryRunCommand:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/apiclient/wait.go#L136-L155
initial timeout was 40s.

> Perhaps a safe timeout duration is 4 minutes until the custom configuration is supported. WDYT?

SGTM, i will update.
we need to make sure these timeouts are printed to the user, before the wait starts.

Reply:
updated,

i pushed changes on top two changes:
https://github.com/kubernetes/kubernetes/compare/bb4c6a21e33e0295dc838a8344e11462de6008d5..32fbb23f3b3bfdd62bbfc5e1ef034c1ea36f0a00
https://github.com/kubernetes/kubernetes/compare/5c9fdb9a93ed275399d7d1358982c7b8b6352199..bb4c6a21e33e0295dc838a8344e11462de6008d5

kubelet timeout is now 4 minutes.
the idea is to show one message that kubelet will boot pods now, and then show two wait messages for kubelet and api-server (both with timeout).

i removed the timeout summing, because later we could wait for more components:
https://github.com/kubernetes/kubernetes/pull/119598
the summing can be re-added, but i don't like showing the user this max value.